cpp/src/parquet/encryption/key_management_test.cc

```diff
@@ -323,6 +323,36 @@ TEST_F(TestEncryptionKeyManagement, KeyRotationWithInternalMaterial) {
   EXPECT_THROW(this->RotateKeys(double_wrapping, encryption_no), ParquetException);
 }
 
+TEST_F(TestEncryptionKeyManagement, UsePropertiesAfterCrytoFactoryDestroyed) {
+  std::shared_ptr<KmsClientFactory> kms_client_factory =
+      std::make_shared<TestOnlyInMemoryKmsClientFactory>(true, key_list_);
```

Comment:
```suggestion
      std::make_shared<TestOnlyInMemoryKmsClientFactory>(/*wrap_locally=*/true, key_list_);
```

Or should we directly use `wrap_locally_`?

Reply:
`wrap_locally_` is only initialized by `SetupCryptoFactory`, which I'm purposely not using in this test due to using a local CryptoFactory instead, so your suggested change makes sense I think.

Reply:
I just noticed I'm using `wrap_locally_` lower down in this function to build the file name though, which doesn't make sense. So I've introduced a local `wrap_locally` instead.